src/test/java/com/users/service/UserServiceTest.java

```diff
@@ -0,0 +1,244 @@
+package com.users.service;
+
+import com.users.criteria.UserCriteriaEasy;
+import com.users.dto.UserDto;
+import com.users.entities.User;
+import com.users.exceptions.EntityAlreadyExistsException;
+import com.users.exceptions.EntityNotFoundException;
+import com.users.mappers.UserMapper;
+import com.users.repository.UserRepository;
+import com.users.utility.UserUtilityTest;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
```

Comment:
la ce te ajuta adnotarea asta?

Reply:
initializeaza automat mock urile si le injecteaza in obiectul adnotat cu @InjectMocks si verifica daca in testele mele exista cod nefolosit sau care nu functioneaza cum ar trebui

Reply:
asta n am inteles  verifica daca in testele mele exista cod nefolosit sau care nu functioneaza cum ar trebui

Reply:
am cautat din nou si am gasit asta:
- Inițializează obiectele mock definite în test prin adnotarea @Mock.
- Permite folosirea adnotării @InjectMocks pentru a injecta automat mock-urile în obiectul testat.
- Simplifică configurarea testelor unitare, eliminând necesitatea de a apela manual anumite metode
Dar am gasit si faptul ca verifica daca toate Mockurile au fost utilizate, iar in cazul in care exista mock uri neutilizate poate arunca o exceptie pentru a curata codul de mockuri inutile